app-text/texlive_texmf/additional-files/createSubpackageInfos.pl

```diff
@@ -0,0 +1,95 @@
+#!/bin/perl
+
+# Helper script to generate the package metadata definitions for the texlive_texmf subpackages
+# usage:
+# 	perl -I $sourceDir/tlpkg createSubpackageInfos.pl <tlpdbroot> > subpackageInfos.recipe
+#
+#	<tlpdbroot> is the root of the TeXLive package database, containing tlpkg/texlive.tlpdb.
+#		It also needs the texmf-dist directory to work.
+#		After these have been symlinked to $sourceDir (by the BUILD function in the recipe),
+#		$sourceDir can be used inside the haikuporter chroot.
+#
+# This script uses the TeXLive perl modules, therefore it needs $sourceDir/tlpkg in @INC.
+
+use strict;
+use warnings;
+
+use TeXLive::TLPDB;
+
+print "# This file has been generated by createSubpackageInfos.pl\n\n";
+
+my $tlpdb = TeXLive::TLPDB->new(root => $ARGV[0]);
+my @collections = $tlpdb->collections();
+
+for my $collectionPkgName (@collections) {
+	my $collection = substr $collectionPkgName, 11;
+	my $collectionPkg = $tlpdb->get_package($collectionPkgName);
+	my $shortdesc = $collectionPkg->shortdesc();
+	my $longdesc = $collectionPkg->longdesc();
+	if (length $shortdesc >= 44) { # SUMMARY must not exceed 80 characters in total
+		$shortdesc = substr($shortdesc, 0, 42) . "…";
+	}
+	if ($shortdesc =~ /\.$/) { # SUMMARY must not end in a '.'
+		$shortdesc = substr($shortdesc, 0, -1);
+	}
+	$shortdesc =~ s'`'\`'g; # escape '`' characters
+	$longdesc =~ s'`'\`'g if $longdesc;
+	my @depends = $collectionPkg->depends();
+	my @runfiles;
+	my @docfiles;
+	my @srcfiles;
+	for my $depName (@depends) {
+		my $depPkg = $tlpdb->get_package($depName);
+		push @runfiles, $depPkg->runfiles();
+		push @docfiles, $depPkg->docfiles();
+		push @srcfiles, $depPkg->srcfiles();
+	}
+	if (@runfiles) {
+		print "SUMMARY_$collection=\"TeXLive Collection: $shortdesc\"\n";
+		print "DESCRIPTION_$collection=\"$longdesc\"\n" if $longdesc;
+		print "PROVIDES_$collection=\"\n";
+		print "\ttexlive_texmf_$collection = \$portVersion\n";
```

Comment:
Should we add PROVIDES for the individual TeX packages in the collection here as well, so that other packages can later depend on them instead of just the whole collection.

How should these be called? Should (or can) we introduce a new prefix (e.g. "tex:")?

Note: I would only do this for the runtime files. This is where it makes the most sense.

The generated file would become ca. 4000 lines longer. I have already implemented it locally but not tested yet.